internal/Board.py

```diff
@@ -49,25 +49,34 @@ def current_square(self, piece):
         return self.position.pieces_positions[piece]
 
     def apply_move(self, move):
-        origin_square = self.position.pieces_positions[move.piece]
-        target_piece = move.square.content
+        target_piece = move.destination_square.content
         self.leave_square(move.piece)
         # If the move is a capture, remove the piece on the destination square
-        if move.is_capture():
-            self.game.add_captured_piece(move.square.content, self.position)
-        self.put_piece_on_square(move.piece, move.square.rank, move.square.file)
+        if move.is_capture:
+            self.game.add_captured_piece(target_piece, self.position)
+        self.put_piece_on_square(move.piece, move.destination_square.rank, move.destination_square.file)
         if move.is_castle:
             rook_move = compute_castling_rook_move(move, self.squares)
             self.leave_square(rook_move.piece)
-            self.put_piece_on_square(rook_move.piece, rook_move.square.rank, rook_move.square.file)
+            self.put_piece_on_square(
+                rook_move.piece,
+                rook_move.destination_square.rank,
+                rook_move.destination_square.file
+            )
             rook_move.piece.never_moved = False
+        if move.is_en_passant:
+            # In this very specific case, the capture square is NOT the landing square
+            captured_pawn_square = self.squares[(move.origin_square.rank, move.destination_square.file)]
+            self.game.add_captured_piece(captured_pawn_square.content, self.position)
+            captured_pawn_square.empty_content()
         move.piece.never_moved = False
+        self.position.latest_move = move
         self.game.move_history.append(move)
         self.position.color_to_move = move.piece.opposite_color()
         self.position.update_controlled_squares(self.squares)
         self.position.update_legal_moves(self.squares)
         self.determine_check_situation(move)
-        print(move.to_string(origin_square, target_piece))
+        print(move.to_string(target_piece))
```

Comment:
I think you may use logger to display infos in console regarding the type of info, it will allow you to distinct error, warning or info :)